src/remat.jl

```diff
@@ -33,48 +33,46 @@ Combine multiple ReMat with the same grouping variable into a single object.
 amalgamate(reterms::Vector{ReMat{T,S} where S}) where {T} = _amalgamate(reterms,T)
 # constant S
 amalgamate(reterms::Vector{ReMat{T,S}}) where {T,S} = _amalgamate(reterms,T)
+
 function _amalgamate(reterms::Vector, T::Type)
-    if length(reterms) == 1
-        return first(reterms)
+    factordict = Dict{Symbol, Vector{Int}}()
+    for (i, rt) in enumerate(reterms)
+        push!(get!(factordict, rt.trm.sym, Int[]), i)
     end
-
-    trm = first(reterms).trm
-    refs = first(reterms).refs
-    levels = first(reterms).levels
-    cnames =  foldl(vcat,[rr.cnames for rr in reterms])
-    z = foldl(vcat, [rr.z for rr in reterms])
-    wtz = z
-
-    Snew = size(z, 1)
-    # taken straight from modelcols(t::RandomEffectsTerm, d::NamedTuple)
-    m = reshape(1:abs2(Snew), (Snew, Snew))
-    mask = BitMatrix([0 for idx in m])
-    blks = [ [size(rr.λ,1), size(rr.λ,1)] for rr in reterms]
-    for (ii, bb) in enumerate(blks)
-        if ii != 1
-            prev = blks[ii-1][2]
-            blks[ii] = [prev+1,prev+bb[1]]
+    length(factordict) == length(reterms) && return reterms
+    value = ReMat{T}[]
```

Comment:
when looking at the entire function, it's obvious that `value` is the return value and thus the amalgated RE, but it might make sense to give it a better name. `result` isn't really that much more descriptive but I find it a bit more transparent. Maybe `simplified_re`? Naming is hard.

Reply:
I often use the name `value` when constructing the value to be returned by the function.  I am not wedded to the name, though.

Reply:
Then let's do the path of least resistance and not change anything.